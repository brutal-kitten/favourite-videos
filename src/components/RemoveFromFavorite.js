import React, { Component } from 'react';

class RemoveFromFavorite extends Component {

  handleClick = (event) => {

    event.preventDefault();
    this.props.removeFromFavorite(this.props.id);
  }

  render () {

    return(
      <div className="removeFromFav" >
        <button className="buttonRemoveFromFav" type="button" onClick={(event) => this.handleClick(event)}>
          <span className="glyphicon glyphicon-star"></span>
        </button>
      </div>
    )
  }
}

export default RemoveFromFavorite;
