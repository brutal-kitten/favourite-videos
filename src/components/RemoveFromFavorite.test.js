import React from 'react';
import ReactDOM from 'react-dom';
import RemoveFromFavorite from './RemoveFromFavorite.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('should display a button', () => {
  const tree = TestRenderer.create(
    <RemoveFromFavorite />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe ('shoud call handleclick function after click', () => {
  const mockRemoveFromFavorite = jest.fn();
  const mockTrigger = jest.fn();
  const props = {
    removeFromFavorite: mockRemoveFromFavorite,
    trigger: mockTrigger,
    id: "12345"
  };

  const wrapper = shallow(<RemoveFromFavorite {...props}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {},
  });

  it('should call removeFromFavorite function with 1 argument - id', () => {
    expect(mockRemoveFromFavorite.mock.calls.length).toBe(1);
    expect(mockRemoveFromFavorite.mock.calls[0][0]).toBe("12345");
  });

  it('should call trigger function', () => {
    expect(mockTrigger.mock.calls.length).toBe(1);
  });

})
