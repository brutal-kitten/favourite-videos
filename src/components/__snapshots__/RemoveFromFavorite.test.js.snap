// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a button 1`] = `
<div
  className="removeFromFav"
>
  <button
    className="buttonRemoveFromFav"
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="glyphicon glyphicon-star"
    />
  </button>
</div>
`;
